models/database.js

```diff
@@ -43,13 +43,18 @@ Database.prototype.all = function all(callback) {
 
     callback(err, rows);
     db.close();
-    statement.finalize();
   });
 }
 
 // OPTIMIZE / TODO: this can only search by one parameter at a time, and only with an `=` relationship.
 Database.prototype.findBy = function findBy(parameter, value, callback) {
-  // FIXME: it'd be nice to have a security check that the parameter is a valid parameter (e.g. make sure it's not sql injection)
+  // check that the parameter is a valid parameter (e.g. make sure it's not sql injection)
+  if (!this._validParam(parameter)) {
+    console.log('!!!!ERROR!!!! In Database#findBy. Unrecognized parameter: ' + parameter);
+    callback({ message: 'Error: syntax error. Unrecognized parameter.', errno: null, code: null }, null);
+    return;
+  }
+
```

Comment:
Should this if statement also be in sortBy since it also takes a parameter that is a column name?